### ISSUE TITLE
Toggle gridster resizable or draggable on individual item? 

### ISSUE THREAD

**Issue by odm3** [ManifestWebDesign/angular-gridster#335]
Is there a way in angular-gridster to make an individual gridster item resziable or draggable or vice versa? 

For example I have a minimize feature for the gridster items on my app. When an individual item is minimized, I don't want it to be able to be resized. Is there a way to specify it on the item level?

**Comment by rbelmega** [comment]
u can do it with css, but I'm not sure that this is the best idea

**Comment by tangorri** [comment]
I actually do this using css it works fine for me. Why is it a bad idea @rbelmega  ?

**Comment by rbelmega** [comment]
@tangorri I guess u still will have this items in your DOM:

< div class="gridster-item-resizable-handler handle-s"></div>
< div class="gridster-item-resizable-handler handle-e"></div>
< div class="gridster-item-resizable-handler handle-n"></div>

which also keep all eventListener handlers
yep, it works well, but we should keep our code clean

**Comment by tangorri** [comment]
Correct @rbelmega that's not an issue for me. It's just may fit @odm3 need. To make it part of the project I would add a param for 'resizable'  n in options mapping.

**Comment by VhatAmI** [comment]
I also would love to be able to make individual widgets resizable (or not) 

Right now its either all or none.

**Comment by VhatAmI** [comment]
@odm3 my workaround was to use ng-class to add a class to the widgets of the type I wanted to not be re-sizable. Resize was set to true for all widgets, but I took gridster-item-resizable-handler and my other class and had a style that hid any tag that inherited both.  It is working pretty well for me so far.

**Comment by AlessandroFlati** [comment]
Just a few minor improvement to default src.
Inside of GridsterDraggable directive, modify this lines (CUSTOM is what you have to add):
```javascript
for (var h = 0, hl = $dragHandles.length; h < hl; ++h) {
    if(itemOptions.draggable !== false) { // CUSTOM
        unifiedInputs[h] = new GridsterTouch($dragHandles[h], mouseDown, mouseMove, mouseUp);
        unifiedInputs[h].enable();								
    }  // CUSTOM
}
```
Same for GridsterResizable directive:
```javascript
for (var c = 0, l = handlesOpts.length; c < l; c++) {
    if(itemOptions.resizable !== false) handles.push(new ResizeHandle(handlesOpts[c])); // CUSTOM
}
```
Then simply add to the desired objects in ```javascript $scope.gridsterItems ``` appropriate keys/values
```javascript 
$scope.gridsterItems = [
    {
        draggable : false // optional
        resizable : false // optional
        ...
    }
]
```

**Comment by rbelmega** [comment]
@AlessandroFlati that's a nice workaround! but modifying source code of lib isn't a good idea (u might have to get a new version or load it from cdn etc...)
it would be nice if u create a pr with ur solution or contributors of this lib will add these feature at next releases

**Comment by AlessandroFlati** [comment]
I'm actually pretty new to github world, but you're right: this should be implemented by ManifestWebDesign themselves.
Tell me how to create a pr and I'll gladly do it.

**Comment by VhatAmI** [comment]
@AlessandroFlati that code is apply-able to individual widgets? That is pretty impressive.

**Comment by AlessandroFlati** [comment]
@VhatAmI Yep. Obviously it's simply a workaround, not a full developed integration to the original plugin (e.g. new properties `draggable` and `resizable` should be set to default `true` on the directive config, but as a matter of fact, the new lines only check if they assume `false` value, so if they are `undefined` the plugin proceeds as usual).
But it does what we want: doesn't create the <div> resizing handlers, doesn't create the listeners and it works on touch devices too. :smile:

**Comment by PedroF20** [comment]
I tried this but it doesn't work. I mean, the draggable attribute changes but the item can still be dragged

**Comment by AlessandroFlati** [comment]
Don't know what is not working on your specific issue. Please make sure you're doing all of the steps in my reply.

**Comment by PedroF20** [comment]
I added those custom lines you published. The attribute changes but it is still draggable...

**Comment by PedroF20** [comment]
`$scope.widgets = [{type: 'area', draggable: true, sizeX: 2, sizeY: 2}];`
This is in my controller

In my view I have a click event in which I change the `draggable`
```

<div gridster="gridsterOpts">
    <ul id="gridster-ul">
        <li ng-repeat="widget in widgets" data-draggable="widget.draggable" gridster-item size-x="widget.sizeX" size-y="widget.sizeY">
            <div class="box-header">
               <h3>Chart</h3>
                <div class="box-header-btns pull-right">
                    <a title="Pin" ng-click="widget.draggable=false">
                        <span class="glyphicon glyphicon-pushpin"></span>
                    </a>
```

**Comment by AlessandroFlati** [comment]
I think you should just remove that strange
`data-draggable="widget.draggable"`.
If it doesn't work, I don't know exactly where to put my hands without seeing the entire code. But I don't see the meaning of that attribute.

**Comment by PedroF20** [comment]
I removed it. The behavior is exactly the same. Maybe I'm doing something wrong with the includes after the src modification?

Anyways, my code is here: [](https://github.com/PedroF20/angular-tracemysteps)

Any insight would be helpful. Thank you.

**Comment by AlessandroFlati** [comment]
Can't see it right now, but I will keep an eye on you. ;)

**Comment by PedroF20** [comment]
Thank you. Feel free to message when you have the time.
